run_dir/static/js/projects.js

```diff
@@ -129,13 +140,10 @@ function load_table_head(columns){
 }
 
 
-
-
-// Undefined columns handled here
-function load_undefined_columns() {
-  return $.getJSON("/api/v1/projects_fields?undefined=true&project_list=" + projects_page_type, function(data) {
+// Undefined columns handled here now
+function load_undefined_columns(cols) {
     var columns_html = "";
-    $.each(data, function(column_no, column) {
+    $.each(cols, function(col_id, column) {
       $("#undefined_columns").append('<div class="checkbox">'+
           '<label>'+
             '<input type="checkbox" class="filterCheckbox" data-columngroup="undefined-columns" data-displayname="'+column+'" name="'+column+'" id="undefined-columns-'+column+'">'+
```

Comment:
the other way around is a foreach in javascript, that amazingly does the
SAME BLOODY STUPID THING

On Fri, Mar 20, 2015 at 4:41 PM, Phil Ewels <notifications@github.com>
wrote:

> In run_dir/static/js/projects.js
> <https://github.com/SciLifeLab/genomics-status/pull/181#discussion_r26849397>
> :
>
> >      var columns_html = "";
> > -    $.each(data, function(column_no, column) {
> > +    $.each(cols, function(col_id, column) {
> >        $("#undefined_columns").append('<div class="checkbox">'+
> >            '<label>'+
> >              '<input type="checkbox" class="filterCheckbox" data-columngroup="undefined-columns" data-displayname="'+column+'" name="'+column+'" id="undefined-columns-'+column+'">'+
>
> $.each(cols, function(whydoineedthisstupidindexcountihatejavascriptsomuch, column) {
>   ...
> });
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/SciLifeLab/genomics-status/pull/181/files#r26849397>.
>